common-npm-packages/docker-common-v2/registryauthenticationprovider/acrauthenticationtokenprovider.ts

```diff
@@ -1,8 +1,11 @@
 "use strict";
 
+import AuthenticationTokenProvider from "./authenticationtokenprovider";
+import { ApplicationTokenCredentials } from "azure-pipelines-tasks-azure-arm-rest-v2/azure-arm-common";
+import RegistryAuthenticationToken from "./registryauthenticationtoken";
 import * as tl from "azure-pipelines-task-lib/task";
-import RegistryAuthenticationToken from "./registryauthenticationtoken"
-import AuthenticationTokenProvider from "./authenticationtokenprovider"
+import * as Q from "q";
```

Comment:
Any reason you can't use async / await pattern for new code and have to fallback to Q ?